std/mem.zig

```diff
@@ -254,6 +255,27 @@ pub fn eql(comptime T: type, a: []const T, b: []const T) bool {
     return true;
 }
 
+/// Compares two slices and returns whether they are equal in constant time.
+pub fn constant_time_eql(comptime T: type, a: []const T, b: []const T) T {
+    //TODO: https://github.com/ziglang/zig/issues/640
+    return @noInlineCall(noinline__constant_time_eql, T, a, b);
+}
+
+fn noinline__constant_time_eql(comptime T: type, a: []const T, b: []const T) T {
+    if (a.len != b.len) return 1;
+    var tmp: T = 0;
+    var count: usize = a.len - 1;
+    while (count != 0) : (count -= 1) {
+        tmp |= a[count] ^ b[count];
```

Comment:
These loads may leak timing information for some instances of T when a or b are or aren't naturally aligned for T, since unaligned loads are usually much slower. The simple workaround is to only accept u8.

Reply:
In this code, all members of `a` and `b` are guaranteed by the type system to be naturally aligned. Slices whose pointers are under- or over-aligned have an `align` annotation, e.g.: `a: []align(1) const T`